db_test.go

```diff
@@ -134,6 +134,49 @@ func TestInsert(t *testing.T) {
 	})
 }
 
+func TestSaveExistingDocument(t *testing.T) {
```

Comment:
I would merge the two tests into a single `TestSaveDocument` test. As a matter of fact, saving a new document would simply require to use `Save()` also at line 148 instead of `InsertOne` and subsequently checking the size of the collection.